src/ui/drawing/ISurface.hh

```diff
@@ -72,10 +73,47 @@ public:
     /// <param name="pImage">The image to draw.</param>
     virtual void DrawImage(int nX, int nY, int nWidth, int nHeight, const ImageReference& pImage) = 0;
 
+    /// <summary>
+    /// Draws a secondary surface onto the surface.
+    /// </summary>
+    /// <param name="nX">The x coordinate to draw at.</param>
+    /// <param name="nY">The y coordinate to draw at.</param>
+    /// <param name="pImage">The surface to draw.</param>
+    virtual void DrawSurface(int nX, int nY, const ISurface& pSurface) = 0;
+
+    /// <summary>
+    /// Sets the alpha value of anything that's not currently transparent to the specified value.
+    /// </summary>
+    /// <param name="nAlpha">The new opacity (1-255).</param>
+    virtual void SetOpacity(UINT8 nAlpha) = 0;
+
 protected:
     ISurface() noexcept = default;
 };
 
+class ISurfaceFactory
+{
+public:
+    virtual ~ISurfaceFactory() = default;
+    ISurfaceFactory(const ISurfaceFactory&) noexcept = delete;
+    ISurfaceFactory& operator=(const ISurfaceFactory&) noexcept = delete;
+    ISurfaceFactory(ISurfaceFactory&&) noexcept = delete;
```

Comment:
I'd need to double check this, but I think this move constructor needs to be protected because it's used in a `unique_ptr`. If it still works with this deleted, don't worry about it.

Reply:
The move constructor is being deleted from all of the existing "interface" classes.